### PR TITLE
refactor(frontend/expr): unifiy `to_expr` and `as_expr`, remove redundent code

### PR DESCRIPTION
**Pull request by Enter-tainer**
`Condition::as_expr` and `Condition::to_expr` are very similar, except for one is for borrowed value and the other one is for owned value. However, these two functions are implemented by copying and pasting code. This pr unifies the borrowed version and the owned version.

## Checklist

- [x] I have written the necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

File: src/frontend/src/expr/utils.rs [hunk 1]
```diff
@@ -12,32 +12,51 @@
...
-                to_conjunctions_inner(expr, rets);
+                split_expr_by(expr, op, rets);
             }
         }
         _ => rets.push(expr),
     }
 }
 
-/// give a bool expression, and transform it to Conjunctive form. e.g. given expression is
+pub fn merge_expr_by<'a, I>(mut exprs: I, op: ExprType, identity_elem: bool) -> ExprImpl
```
Comment (st1page) [inline src/frontend/src/expr/utils.rs:36]: maybe
```suggestion
pub fn merge_expr_by_binary<'a, I>(mut exprs: I, op: ExprType, identity_elem: ExprImpl) -> ExprImpl
```

File: src/frontend/src/utils/condition.rs [hunk 2]
```diff
@@ -74,49 +74,29 @@ impl Condition {
...
-                ret = FunctionCall::new(ExprType::And, vec![ret, expr.clone()])
-                    .unwrap()
-                    .into();
-            }
-            ret
-        } else {
-            Literal::new(Some(ScalarImpl::Bool(true)), DataType::Boolean).into()
-        }
+        merge_expr_by_binary(
+            self.conjunctions.iter().map(Cow::Borrowed),
```
Comment (BugenZhao) [inline src/frontend/src/utils/condition.rs:88]: Why not directly `cloned()` here? Seems we are always calling `into_owned` on `Cow`s.

File: src/frontend/src/utils/condition.rs [hunk 3]
```diff
@@ -74,49 +74,29 @@ impl Condition {
     }
 
     pub fn to_expr(self) -> ExprImpl {
```
Comment (BugenZhao) [inline src/frontend/src/utils/condition.rs:76]: Suggest naming as `into_expr`.

File: src/frontend/src/utils/condition.rs [hunk 4]
```diff
@@ -74,49 +74,29 @@ impl Condition {
...
+        merge_expr_by_binary(
+            self.conjunctions.into_iter().map(Cow::Owned),
+            ExprType::And,
+            ExprImpl::literal_bool(true),
+        )
     }
 
     // TODO(TaoWu): We might also use `Vec<ExprImpl>` form of predicates in compute node,
     // rather than using `AND` to combine them.
     pub fn as_expr(&self) -> ExprImpl {
```
Comment (BugenZhao) [inline src/frontend/src/utils/condition.rs:86]: And this one `to_expr`.

File: src/frontend/src/utils/condition.rs (hunk 3 above)
Comment (xxchan) [reply src/frontend/src/utils/condition.rs]: Then why not `impl Into<ExprImpl> for Condition`

File: src/frontend/src/utils/condition.rs [hunk 5]
```diff
@@ -73,50 +72,30 @@ impl Condition {
...
-            for expr in iter {
-                ret = FunctionCall::new(ExprType::And, vec![ret, expr])
-                    .unwrap()
-                    .into();
-            }
-            ret
-        } else {
-            Literal::new(Some(ScalarImpl::Bool(true)), DataType::Boolean).into()
-        }
+    pub fn into_expr(self) -> ExprImpl {
```
Comment (xxchan) [inline src/frontend/src/utils/condition.rs:75]: What about `impl Into<ExprImpl> for Condition`?

Comment (Enter-tainer) [reply src/frontend/src/utils/condition.rs:75]: How about `impl From<Condition> for ExprImpl` and `impl From<'a &Condition> for ExprImpl` ?

**Review by st1page — APPROVED** [review]
LGTM

**Review by xxchan — APPROVED** [review]

**Pull request merged**